AutoRest/Generators/Ruby/Azure.Ruby/AzureRubyCodeGenerator.cs

```diff
@@ -58,6 +66,78 @@ public override void NormalizeClientModel(ServiceClient serviceClient)
             AzureExtensions.NormalizeAzureClientModel(serviceClient, Settings, CodeNamer);
             CorrectFilterParameters(serviceClient);
             base.NormalizeClientModel(serviceClient);
+            AddRubyPageableMethod(serviceClient);
+            ApplyPagination(serviceClient);
+
+        }
+
+        /// <summary>
+        /// Adds method to use for autopagination
+        /// </summary>
+        /// <param name="serviceClient">The service client.</param>
+        private void AddRubyPageableMethod(ServiceClient serviceClient)
+        {
+            if (serviceClient == null)
+            {
+                throw new ArgumentNullException("serviceClient");
```

Comment:
In general my preferece'd be to use Nameof operator instead of string like throw new ArgumentNullException(nameof(serviceClient)); so that just incase we change the name of parameter it'll force us to change this line as well.